test_xpartition.py

```diff
@@ -44,15 +117,18 @@ def da(request):
     return _construct_dataarray(shape, chunks, name)
 
 
-def test_block_indices(da):
-    blocks = xpartition.block_indices(da)
-    blocks_shape = tuple(blocks.sizes[dim] for dim in da.dims)
-    stacked_blocks = blocks.stack(block=[dim for dim in da.dims])
-    for i in range(stacked_blocks.sizes["block"]):
-        indexers = xpartition.block_to_slices(stacked_blocks.isel(block=i))
-        dask_blocks_indices = np.unravel_index(i, blocks_shape)
-        block_data_via_xarray = da.isel(indexers).data.compute()
-        block_data_via_dask = da.data.blocks[dask_blocks_indices].compute()
+def test_indexers_with_scalars(da):
```

Comment:
Yeah that's a good idea.  I added more rigorous tests for `blocks.indexers` and removed the tests of the private functions; since `blocks.isel` is just a thin wrapper around `blocks.indexers`, I kept its test fairly minimal.